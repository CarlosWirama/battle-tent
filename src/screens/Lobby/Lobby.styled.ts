import styled from 'styled-components';
import { Centered } from '../../components/basics';
import Background from '../../assets/images/ui/lobby-background.jpg';

export const LayoutContainer = styled(Centered)`
  background-image: linear-gradient(141deg,#ffab63 0%,#46bdca 51%,#4eb2d4 75%);
  background-image: url(${Background});
  height: 100vh;
  color: white;
`;
