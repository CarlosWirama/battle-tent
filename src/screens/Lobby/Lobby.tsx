import React, { useState, useContext } from 'react';
import { Input, Button } from '../../components/basics';
import Modal, { ModalProps } from '../../components/Modal';
import LoadingIndicator from '../../components/LoadingIndicator';
import { NavigationContext, ScreenState } from '../../navigation';

export default function Lobby() {
  const [inviteModalShown, setInviteModalShown] = useState(false);
  const [isLoading, setIsLoading] = useState(false);
  const navigate = useContext(NavigationContext);

  async function onFindMatch() {
    setIsLoading(true);
    try {
      const response = await fetch('');
      if (response.status !== 200) throw response;
      setTimeout(() => navigate(ScreenState.Battle), 1500)
    } catch (e) {
      console.error(e);
      alert(e);
    }
    // setIsLoading(false);
    setTimeout(() => setIsLoading(false), 1500)
  }
  async function onInviteMatch() {
    setIsLoading(true);
    try {
      const response = await fetch('');
      if (response.status !== 200) throw response;
    } catch (e) {
      console.error(e);
      alert(e);
    }
    setIsLoading(false);
  }

  function openInviteModal() {
    setInviteModalShown(true);
    onInviteMatch();
  }
  function closeInviteModal() {
    setInviteModalShown(false);
  }

  return (
    <div>
      <InviteModal shown={inviteModalShown} onClose={closeInviteModal} />
      <header>
        <h1>Battle Tent</h1>
      </header>
      {isLoading ? <LoadingIndicator /> :
        <div>
          <Button onClick={onFindMatch}>Find match</Button>
          <Button disabled onClick={openInviteModal}>Invite</Button>
        </div>
      }
    </div>
  );
}

function InviteModal({ shown, onClose }: Omit<ModalProps, 'children'>) {
  return (
    <Modal shown={shown} onClose={onClose}>
      Who do you want to invite?
      <Input />
    </Modal>
  );
}
