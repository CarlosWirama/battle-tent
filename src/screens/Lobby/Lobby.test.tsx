import React from 'react';
import { render } from '@testing-library/react';
import Lobby from './Lobby';

describe('Lobby screen', () => {
  it('should render', () => {
    const { asFragment } = render(<Lobby />);
    expect(asFragment()).toMatchSnapshot();
  });
});
