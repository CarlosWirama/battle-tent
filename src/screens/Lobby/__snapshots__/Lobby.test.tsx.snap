// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lobby screen should render 1`] = `
<DocumentFragment>
  <div
    class="sc-AxirZ sc-fzplWN iSSJWi"
  >
    <header>
      <h1>
        Battle Tent
      </h1>
    </header>
    <div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text sc-AxjAm faKiNC"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Find match
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text sc-AxjAm faKiNC Mui-disabled Mui-disabled"
        disabled=""
        tabindex="-1"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Invite
        </span>
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text sc-AxjAm faKiNC"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Change name
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
